target/scripts/helpers/log.sh

```diff
@@ -116,4 +108,34 @@ function _log
   fi
 }
 
-function _log_with_date { _log "${1}" "$(date '+%Y-%m-%d %H:%M:%S')  ${2}" ; }
+# Like `_log` but adds a timestamp in front of the message.
+function _log_with_date
+{
+  _log "${1}" "$(date '+%Y-%m-%d %H:%M:%S')  ${2}"
+}
+
+# Get the value of the environment variable LOG_LEVEL if
+# it is set. Otherwise, try to query the common environment
+# variables file. If this does not yield a value either,
+# use the default log level.
+function _get_log_level_or_default
+{
+  if [[ -n ${LOG_LEVEL+set} ]]
+  then
+    printf '%s' "${LOG_LEVEL}"
+  elif [[ -e /etc/dms-settings ]]
+  then
+    local LOG_LEVEL
+    LOG_LEVEL=$(grep "^LOG_LEVEL=" /etc/dms-settings | cut -d '=' -f 2)
+    printf '%s' "${LOG_LEVEL:1:-1}"
```

Comment:
Just to clarify, neither of these are setting a fallback value of `1`/`-1` right? This is bash syntax for truncating left/right to remove wrapped quotes of the value stored in `/etc/dms-settings`?

Is there a situation where this ever makes sense to support?

Your last `else` condition is to use the same default `info` as the startup scripts assign via `variables.sh` for a fallback?

The original addition of `/etc/dms-settings` was for a Dovecot quota test that needed to assume to workaround an ENV issue IIRC. In that situation, there isn't a fallback and the ENV may not be set but be assigned a default during startup that could be checked with `/etc/dms-settings`.

Reply:
> Just to clarify, neither of these are setting a fallback value of `1`/`-1` right? This is bash syntax for truncating left/right to remove wrapped quotes of the value stored in `/etc/dms-settings`?

This syntax look very much like the "assign a default value" syntax, I agree, but you are correct, this is _not_ setting any default :) It is just for truncating quotes. @casperklein mentioned that `| cut -f "'" -f 2` would also work. I am fine with both, but maybe you prefer the alternative @polarathene?

> Is there a situation where this conditional for `/etc/dms-settings` ever makes sense to support?
> Resolved (answer: potentially...?)

I guess when you import the `log.sh` helper in a custom script without sourcing `etc/dms-settings`. I was unable to identify a situation where it would do harm though :)